Lib/test/_test_multiprocessing.py

```diff
@@ -4440,6 +4440,31 @@ def test_shared_memory_cleaned_after_process_termination(self):
                     "resource_tracker: There appear to be 1 leaked "
                     "shared_memory objects to clean up at shutdown", err)
 
+    @unittest.skipIf(os.name != "posix", "resource_tracker is posix only")
+    def test_shared_memory_untracking(self):
+        # gh-82300: When a separate Python process accesses shared memory
+        # with track=False, it must not register with the resource tracker.
```

Comment:
Which part of this test would fail if `track=False` did not work properly?

Reply:
The sole effect of `track=False` is that `register` is not called. Thus it would be faulty behavior if `register` is called when `track=False`. The function checks the error message from the subprocess and will fail when that happens.

Reply:
Rather than relying on mocking, I think it would be better to test the actual situation that we're trying to fix:
1) create shared memory segment in parent
2) pass its name to child that will reference it and then close it
3) check that the segment can still be used in parent

Reply:
The main concern I had with that approach is the indeterminate amount of time between closing the process and the resource tracker kicking in. Right now such a time is not needed because the test waits until stderr is fully available and that includes the resource tracker itself. But that's only due to the way the resource tracker is created. If we do rely on that internal behavior then the current test would be in the same vein. Otherwise, what would a good delay be? A second?

Reply:
Hmm, that's a good point and that complicates things a bit...

Reply:
I think technically the previous test guarantees that but I have now added both the `track=False` and `track=True` cases explicitly. The second case needs something from stderr and so it would fail if the resource tracker did not use the same pipe.